thirdparty/github.go

```diff
@@ -352,65 +362,38 @@ func parseGithubErrorResponse(resp *github.Response) error {
 // GetGithubFile returns a struct that contains the contents of files within
 // a repository as Base64 encoded content. Ref should be the commit hash or branch (defaults to master).
 func GetGithubFile(ctx context.Context, token, owner, repo, path, ref string) (*github.RepositoryContent, error) {
+	caller := "GetGithubFile"
+	ctx, span := tracer.Start(ctx, caller, trace.WithAttributes(
+		attribute.String(githubEndpointAttribute, caller),
+		attribute.String(githubOwnerAttribute, owner),
+		attribute.String(githubRepoAttribute, repo),
+		attribute.String(githubRefAttribute, ref),
+		attribute.String(githubPathAttribute, path),
+	))
+	defer span.End()
+
+	if path == "" {
+		return nil, errors.New("remote repository path cannot be empty")
+	}
+
+	githubClient := getGithubClient(getInstallationToken(ctx, token, owner, repo, nil), caller, retryConfig{retry: true})
+
 	var opt *github.RepositoryContentGetOptions
 	if len(ref) != 0 {
 		opt = &github.RepositoryContentGetOptions{
 			Ref: ref,
 		}
 	}
 
-	installationToken, err := GetInstallationToken(ctx, owner, repo, nil)
-	if err != nil {
-		ctx, httpClient, putClient := getGithubClient(ctx, installationToken, "GetGithubFile", retryConfig{retry: true}, []attribute.KeyValue{
-			attribute.String(githubOwnerAttribute, owner),
-			attribute.String(githubRepoAttribute, repo),
-			attribute.String(githubRefAttribute, ref),
-			attribute.String(githubPathAttribute, path),
-		})
-		defer putClient()
```

Comment:
@bynn this reworks the github client code so you may want to review it to see how it impacts EVG-19966.